src/simulation/hdf_simulation_store.jl

```diff
@@ -50,7 +51,7 @@ mutable struct HdfSimulationStore <: SimulationStore
     # The key is the problem name.
     optimizer_stats_datasets::Dict{Symbol, HDF5.Dataset}
     optimizer_stats_write_index::Dict{Symbol, Int}
-    cache::ResultCache
+    cache::OptimizationOutputCache
```

Comment:
We are using the terms `read_result`, `write_result`, `OutputCache`, `OptimizationOutputCache`, `OutputCacheKey`.

`result` and `output` are referring to the same thing, and so we could use the same term. Either is OK, but I think `result` is more specific.

Take this example:
```
add_result!(output_cache, ...)
```
could be
```
add_result!(result_cache, ...)
# or
add_output!(output_cache, ...)
```
 What do you think?

Reply:
We need to distinguish the functions with the word `result` that imply an interface to the simulation results (post-processed) and the outputs of the optimizer. If we change the names from result to output we would be implying that the interface to the store is for that purpose and not direct result gathering.

Reply:
resolved in 
4fbead9